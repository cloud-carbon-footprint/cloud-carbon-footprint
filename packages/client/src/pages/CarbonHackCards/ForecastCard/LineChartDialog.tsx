import * as React from 'react'
import Box from '@mui/material/Box'
import InfoIcon from '@material-ui/icons/Info'
import Typography from '@mui/material/Typography'
import Modal from '@mui/material/Modal'
import { ForecastLineChart } from './ForecastLineChart'
import moment from 'moment'

const style = {
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: 1000,
  bgcolor: 'background.paper',
  border: '2px solid #000',
  boxShadow: 24,
  p: 4,
  borderRadius: '25px',
}

export default function LineChartDialog({ testId, forecastData, region }) {
  const [open, setOpen] = React.useState(false)
  const handleOpen = () => setOpen(true)
  const handleClose = () => setOpen(false)
  const data = []
  const categories = []

  const convertUTCtoLocalTime = (utcTime) => {
    const stillUtc = moment.utc(utcTime).toDate()
    const local = moment(stillUtc).local().format('YYYY-MM-DD HH:mm:ss')
    return local
  }

  forecastData.map((forecast) => {
    data.push(forecast.value)
    categories.push(convertUTCtoLocalTime(forecast.timestamp))
  })

  return (
    <div data-testid={testId}>
      <InfoIcon onClick={handleOpen} data-testid="info-icon"></InfoIcon>
      <Modal
        open={open}
        onClose={handleClose}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        <Box sx={style}>
          <Typography
            data-testid="modal-modal-title"
            variant="h6"
            component="h2"
          >
            Forecast for {region}
          </Typography>
          <ForecastLineChart
            data={data}
            categories={categories}
          ></ForecastLineChart>
        </Box>
      </Modal>
    </div>
  )
}
