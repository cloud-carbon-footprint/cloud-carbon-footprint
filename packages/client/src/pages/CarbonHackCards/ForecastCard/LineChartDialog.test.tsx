import { fireEvent, render, within } from '@testing-library/react'
import React from 'react'
import { create } from 'react-test-renderer'
import LineChartDialog from './LineChartDialog'

jest.mock('./ForecastLineChart')

describe('LineChartDialog', () => {
  const mockInputData = [
    {
      duration: 60,
      location: 'CAISO_NORTH',
      timestamp: '2022-11-09T07:25:00+00:00',
      value: 435.0320147282777,
    },
    {
      duration: 60,
      location: 'CAISO_NORTH',
      timestamp: '2022-11-09T07:30:00+00:00',
      value: 435.07368308252774,
    },
    {
      duration: 60,
      location: 'CAISO_NORTH',
      timestamp: '2022-11-09T07:35:00+00:00',
      value: 435.1770568685555,
    },
  ]

  it('renders line chart dialog', async () => {
    const root = create(
      <LineChartDialog
        testId="line-chart-dialog"
        forecastData={mockInputData}
        region={'us-west-1'}
      />,
    )
    expect(root.toJSON()).toMatchSnapshot()
  })
  it('renders title of the dialog', async () => {
    const { getByTestId, findByText } = render(
      <LineChartDialog
        testId="line-chart-dialog"
        forecastData={mockInputData}
        region={'us-west-1'}
      />,
    )
    fireEvent.click(getByTestId('info-icon'))
    const title = within(await findByText('Forecast for us-west-1'))
    expect(title).not.toBeNull()
  })
})
