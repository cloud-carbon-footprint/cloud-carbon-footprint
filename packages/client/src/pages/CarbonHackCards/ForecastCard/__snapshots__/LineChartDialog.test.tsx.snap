// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChartDialog renders line chart dialog 1`] = `
<div
  data-testid="line-chart-dialog"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    data-testid="info-icon"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
</div>
`;
